src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,603 @@
+        JarDependency(org = 'org.apache.ant', name = 'ant', rev = '1.8.2'),
+        JarDependency(org = 'org.pegdown', name = 'pegdown', rev = '1.4.2'),
+        JarDependency(org = 'org.testng', name = 'testng', rev = '6.8.7'),
+        JarDependency(org = 'org.scalacheck', name = 'scalacheck_2.11', rev = '1.13.1'),
+        JarDependency(org = 'org.jmock', name = 'jmock-legacy', rev = '2.5.1'),
+        JarDependency(org = 'org.easymock', name = 'easymockclassextension', rev = '3.1'),
+        JarDependency(org = 'org.seleniumhq.selenium', name = 'selenium-java', rev = '2.35.0'),
+      ],
+      custom_rules=[
+        Shader.exclude_package('*', recursive=True),]
+    )
+    cls.register_jvm_tool(
+      register,
+      'rsc_outline',
+      classpath=[
+          JarDependency(
+              org='com.twitter',
+              name='rsc_2.11',
+              rev=rsc_outline_version,
+          ),
+          JarDependency(
+              org='com.twitter',
+              name='mjar_2.11',
+              rev=rsc_outline_version,
+          ),
+          JarDependency(
+            org='org.scalameta',
+            name='metacp_2.11',
+            rev=scalameta_version,
+          ),
+          JarDependency(
+            org='org.scalameta',
+            name='metai_2.11',
+            rev=scalameta_version,
+          ),
+      ],
+      custom_rules=[
+        Shader.exclude_package('scala', recursive=True),
+        Shader.exclude_package('rsc', recursive=True),
+      ]
+      )
+
+  def register_extra_products_from_contexts(self, targets, compile_contexts):
+    super(RscCompile, self).register_extra_products_from_contexts(targets, compile_contexts)
+    # TODO when digests are added, if the target is valid,
+    # the digest should be loaded in from the cc somehow.
+    for target in targets:
+      rsc_outline_cc, compile_cc = compile_contexts[target]
+      if target.has_sources('.java'):
+        self.context.products.get_data('rsc_outline_classpath').add_for_target(
+          compile_cc.target,
+          [(conf, compile_cc.jar_file) for conf in self._confs])
+      elif target.has_sources('.scala'):
+        self.context.products.get_data('rsc_outline_classpath').add_for_target(
+          rsc_outline_cc.target,
+          [(conf, rsc_outline_cc.rsc_mjar_file) for conf in self._confs])
+      else:
+        pass
+
+  def create_empty_extra_products(self):
+    super(RscCompile, self).create_empty_extra_products()
+
+    compile_classpath = self.context.products.get_data('compile_classpath')
+    classpath_product = self.context.products.get_data('rsc_outline_classpath')
+    if not classpath_product:
+      self.context.products.get_data('rsc_outline_classpath', compile_classpath.copy)
+    else:
+      classpath_product.update(compile_classpath)
+
+  def execute(self):
+    if JvmPlatform.global_instance().get_options().compiler == 'rsc':
+      # Calling BaseZincCompile directly because ZincCompile won't run a compile if
```

Comment:
Hm. I could change jvm_compile so that it knows about `JvmPlatform.global_instance().get_options().compiler` and add an abstract property for the current tasks' compiler. Then that could be pushed up into jvm compile.